plugins/@grouparoo/google-sheets/src/lib/sheet-export/exportRecords.ts

```diff
@@ -0,0 +1,261 @@
+      rows = await client.read({ limit: GOOGLE_SHEETS_ROWS_LIMIT, offset });
+    }
+  }
+};
+
+// delete the given destinationIds
+const deleteByDestinationIds: BatchMethodDeleteByDestinationIds = async ({
+  client,
+  users,
+}) => {
+  for (const user of users) {
+    // No batch delete available, do one by one
+    try {
+      await client.cleanRowByRowNumber(user.destinationId);
+    } catch (error) {
+      user.error = error;
+    }
+  }
+};
+
+// update these users by destinationId
+const updateByDestinationIds: BatchMethodUpdateByDestinationIds = async ({
+  client,
+  users,
+  config,
+}) => {
+  for (const user of users) {
+    try {
+      const { newRecordProperties, oldRecordProperties } = user;
+      const id = newRecordProperties[config.foreignKey];
+      const oldId = oldRecordProperties[config.foreignKey];
+      if (oldId && oldId !== id && config.syncOperations) {
+        await deleteOrCleanGroups(
+          client,
+          config.foreignKey,
+          oldId,
+          config.destinationOptions?.groupsColumn?.toString(),
+          config.syncOperations.delete
+        );
+      }
+      const payload = buildPayload(config, user);
+      await client.updateRowByRowNumber(user.destinationId, payload);
+    } catch (error) {
+      user.error = error;
+    }
+  }
+};
+
+// usually this is creating them. ideally upsert. set the destinationId on each when done
+const createByForeignKeyAndSetDestinationIds: BatchMethodCreateByForeignKeyAndSetDestinationIds =
+  async ({ client, users, config }) => {
+    for (const user of users) {
+      try {
+        const payload = buildPayload(config, user);
+        const destinationId = await client.addRowAtTheEnd(payload);
+        if (destinationId) {
+          user.destinationId = destinationId;
+        }
+      } catch (error) {
+        user.error = error;
+      }
+    }
+  };
+
+function buildPayload(config: BatchConfig, user) {
+  const { newRecordProperties, oldRecordProperties, oldGroups, newGroups } =
+    user;
+  const deletePropertiesPayload = {};
+  const newPropertyKeys = Object.keys(newRecordProperties);
+  Object.keys(oldRecordProperties)
+    .filter((k) => !newPropertyKeys.includes(k))
+    .forEach((k) => (deletePropertiesPayload[k] = null));
+  let payload = Object.assign(newRecordProperties, deletePropertiesPayload);
+  const formattedDataFields = {};
+  for (const key of Object.keys(payload)) {
+    formattedDataFields[key] = formatVar(payload[key]);
+  }
+  const groupsColumn = config?.destinationOptions?.groupsColumn?.toString();
+  if (groupsColumn) {
+    formattedDataFields[groupsColumn] = getGroupsListToExport(
+      user,
+      groupsColumn,
+      oldGroups,
+      newGroups
+    );
+  }
+  return formattedDataFields;
+}
+
+function getGroupsListToExport(record, groupsColumn, oldGroups, newGroups) {
+  const groups = newGroups;
+  if (record.destinationId && record.result && record.result[groupsColumn]) {
+    const removedGroups = [];
```

Comment:
I think maybe you don't have to go through `oldGroups` here. You just want the new group names

Reply:
In other words, it doesn't seem useful to keep groups that people set by hand in this case.